include/ast.hpp

```diff
@@ -175,6 +175,33 @@ struct RecordVarDeclNode : public DeclNode {
   std::vector<std::unique_ptr<InitExprNode>> inits;
 };
 
+/// @brief An enumeration constant in an enumeration.
+struct EnumConstDeclNode : public DeclNode {
+  EnumConstDeclNode(Location loc, std::string id,
+                    std::unique_ptr<IntConstExprNode> value = {})
+      : DeclNode{loc, std::move(id),
+                 std::make_unique<PrimType>(PrimitiveType::kInt)},
+        value{std::move(value)} {}
+
+  void Accept(NonModifyingVisitor&) const override;
+  void Accept(ModifyingVisitor&) override;
+
+  std::unique_ptr<IntConstExprNode> value;
```

Comment:
```suggestion
  std::unique_ptr<IntConstExprNode> int_const;
```
Better to understand with the following in the parser.
```
enum_const->int_const->val
```